dali/util/image.h

```diff
@@ -98,18 +98,10 @@ void WriteImageScaleBias(const T *img, int h, int w,
   DALI_ENFORCE(w >= 0);
   DALI_ENFORCE(c >= 0);
   CUDA_CALL(cudaDeviceSynchronize());
```

Comment:
Do we need this sync here?

Reply:
I left it because the case where `img`'s wrapping TensorList has some operations scheduled on a non default stream:
e.g.
```
TensorList<GPUBackend> tl;
// tl init
Memcpy(tl, another_tl, non_default_stream)
WriteHWCBatch(tl, "_suff");
```
it is safer to device sync before accessing `tl`'s data on another thread in WriteHWC.

Of course, we could pass a stream to WriteHWCBatch but since WriteHWC is mostly for debug purpose, I think we can get along with `cudaDeviceSynchronize`.

Reply:
Understood